public/app/features/alerting/unified/components/rule-editor/alert-rule-form/simplifiedRouting/route-settings/RouteSettings.tsx

```diff
@@ -37,15 +46,22 @@ export const RoutingSettings = ({ alertManager }: RoutingSettingsProps) => {
     watch,
     register,
     setValue,
+    getValues,
     formState: { errors },
   } = useFormContext<RuleFormValues>();
   const [groupByOptions, setGroupByOptions] = useState(stringsToSelectableValues([]));
-  const { groupIntervalValue, groupWaitValue, repeatIntervalValue } = getDefaultsForRoutingSettings();
+  const { groupIntervalValue, groupWaitValue, repeatIntervalValue } = DEFAULTS_TIMINGS;
   const overrideGrouping = watch(`contactPoints.${alertManager}.overrideGrouping`);
   const overrideTimings = watch(`contactPoints.${alertManager}.overrideTimings`);
-  const requiredFieldsInGroupBy = ['grafana_folder', 'alertname'];
-  const disableGrouping = '...';
+  const groupByCount = getValues(`contactPoints.${alertManager}.groupBy`)?.length ?? 0;
```

Comment:
Shouldn't `watch` be used here? AFAIK using `getValues` might result in `useEffect` having stale values

Reply:
🤔 good point. Actually in my first PR I used watch instead. Let me check it 👀

Reply:
Actually, it doesn't have any wrong effect in this case as, the watch we do before cause a re-render, but I think it makes more sense to use watch as we should not depend on the previous watch's

Reply:
updated!